### PR TITLE
Upload

### PR DESCRIPTION
**Pull request by jdlrobson**

### CODE REVIEW

File: assets/www/js/app.js [hunk 1]
```diff
@@ -70,7 +70,26 @@ require( [ 'jquery', 'l10n', 'geo', 'api', 'templates', 'monuments', 'preference
...
+	}
+	
+	function goBack() {
+		var pageName;
+		if( pageHistory.length > 1 ) {
+			pageName = pageHistory.pop(); // this is the current page
+			pageName = pageHistory.pop(); // this is the previous page
+			showPage( pageName );
+		} else {
+			console.log( 'app.js / goBack: nothing to go back to' );
```
Comment (yuvipanda) [inline assets/www/js/app.js:87]: Don't need this - console.log reports line numbers

**Comment by yuvipanda** [comment]
This doesn't actually cancel the upload at all - just closes the popup. Shouldn't it also cancel the upload?

File: assets/www/js/app.js (hunk 1 above)
Comment (jdlrobson) [inline assets/www/js/app.js:87]: makes sense. Will update

**Comment by jdlrobson** [comment]
@yuvipanda see the commit message - https://github.com/jdlrobson/WLMMobile/commit/c2d0f23cf06f497197ba9cd3bf2c4d1d94e760d4
I wasn't quite sure how to refactor the upload process to allow cancellation and wanted to discuss that with you (the deferred object return is a promise - I wasn't sure if there was a reason for this) and also the upload is two steps and creates a second deferred object - so in fact there are 2 deferreds that need cancelling..

**Comment by yuvipanda** [comment]
@jdlrobson From http://docs.phonegap.com/en/1.9.0/cordova_file_file.md.html#FileTransfer it doesn't look like PhoneGap offers us a way to cancel this :( I think touching this before the Chunked Upload Plugin is done might not be worth it.

**Comment by jdlrobson** [comment]
oohh I got my pull requests confused. yes the cancel button here is the cancel button when you see upload photo or choose from gallery - so it cancels it before you make the reject. The cancel during the upload itself will come in another pull request. See #20 for amended version of this

**Pull request closed without merging**